File: 0062-unique-paths/0062-unique-paths.cpp
class Solution {
public:
    int uniquePaths(int m, int n) {
        if (m == 0 && n == 0) {
            return 0;
        }
        if (m == 1 && n == 1) {
            return 1;
        }
        vector<vector<int>> array(m, vector<int>(n, 1));
        array[0][0] = 0;
        for (int i = 1; i < array.size(); i++) {
            for (int j = 1; j < array[i].size(); j++) {
                array[i][j] = array[i - 1][j] + array[i][j - 1];
            }
        }
        return array[m - 1][n - 1];
    }
};

File: 0062-unique-paths/0062-unique-paths.py
class Solution(object):
    def uniquePaths(self, m, n):
        memo={}
        def recur(m,n):
            if m==1 and n==1:
                return 1
            if m<0 or n<0:
                return 0
            if (m,n) in memo:
                return memo[(m,n)]
            recur(m-1,n)
            recur(m,n-1)
            memo[(m,n)]=recur(m-1,n)+recur(m,n-1)
            return memo[(m,n)]
        answer=recur(m,n)
        return answer